File: Formation.ProjetShop/ProjetShopBoot/src/main/java/ProjetShopBoot/entities/Facture.java
package ProjetShopBoot.entities;

import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Facture {

	@JsonView(JsonViews.Commande.class)
	private double tva;

	@JsonView(JsonViews.Commande.class)
	private TypePaiment typePaiement;

	@JsonView(JsonViews.Commande.class)
	private double remise;

	public Facture() {

	}

	public Facture(double tva, TypePaiment typePaiement, double remise) {
		this.tva = tva;
		this.typePaiement = typePaiement;
		this.remise = remise;
	}

	public double getTva() {
		return tva;
	}

	public void setTva(double tva) {
		this.tva = tva;
	}

	public TypePaiment getTypePaiement() {
		return typePaiement;
	}

	public void setTypePaiement(TypePaiment typePaiement) {
		this.typePaiement = typePaiement;
	}

	public double getRemise() {
		return remise;
	}

	public void setRemise(double remise) {
		this.remise = remise;
	}
}

File: Formation.ProjetShop/ProjetShopAngular/src/app/models/commande/facture.ts
import { Typepaiement } from "../enumerations/TypePaiement";


export class Facture {

  public constructor(
    private _tva?: number,
    private _remise?: number,
    public typePaiement?: Typepaiement,

  ) { }

  public get tva(): number | undefined {
    return this._tva;
  }

  public set tva(value: number | undefined) {
    this._tva = value;
  }


  public get remise(): number | undefined {
    return this._remise;
  }

  public set remise(value: number | undefined) {
    this._remise = value;
  }
  public get paiement(): Typepaiement | undefined {
    return this.typePaiement;
  }
  public set paiement(value: Typepaiement | undefined) {
    this.typePaiement = value;
  }
}

